dir.go

```diff
@@ -280,6 +280,30 @@ func (d *Directory) ListNames(ctx context.Context) ([]string, error) {
 	return out, nil
 }
 
+func (d *Directory) ListEntries(ctx context.Context) ([]FSNode, error) {
+	var entries []FSNode
+
+	names, err := d.ListNames(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, name := range names {
+		child, err := d.Child(name)
+		if err != nil {
+			return nil, err
+		}
+		d.uncache(name)
+		// Remove the cache entry generated by `Child` to avoid using
+		// too much memory.
+		// TODO: Provide a `Child`-like method that doesn't cache the retrieved child.
```

Comment:
Again, this is more than *caching*. It records which files are "open". If an `FSNode` returned from multiple calls to `ListEntries` are used to to modify a file concurrently, we'll end up with two open file descriptors for writing (and each will clobber the other).